src/main/java/io/opentracing/contrib/loggertracer/LoggerSpan.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright 2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.loggertracer;
+
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import java.util.Collections;
+import java.util.Map;
+
+public class LoggerSpan implements Span {
+    private Span wrapped;
+    private final Reporter reporter;
+
+    public final String spanId;
+    public String operationName;
+    public final long startAt;
+    public final Map<String, Object> tags;
+    public final Map<String, String> references;
+
+    public LoggerSpan(Span wrapped, Reporter reporter, String spanId, String operationName, Map<String,Object> tags, Map<String, String> references) {
+        this.reporter = reporter;
+        this.spanId = spanId;
+        this.wrapped = wrapped;
+        this.operationName = operationName;
+        this.tags = tags;
+        this.references = references;
+        startAt = now();
+        reporter.start(startAt, this);
+    }
+
+    protected long now(){
+        return System.nanoTime() / 1000l;
+    }
+
+
+    //------------------------------------------------------------------------------------------------------------------
+    // Span
+
+    @Override
+    public SpanContext context() {
+        return new LoggerSpanContext(wrapped.context(), spanId);
```

Comment:
I wanted to propose to store context as span property but some ot impls create new `SpanContext` when baggage item is added 
https://github.com/uber/jaeger-client-java/blob/master/jaeger-core/src/main/java/com/uber/jaeger/Span.java#L134
https://github.com/uber/jaeger-client-java/blob/master/jaeger-core/src/main/java/com/uber/jaeger/SpanContext.java#L137

you would need to store wrapped span in wrapped context